supervisor/api/resolution.py

```diff
@@ -18,24 +18,40 @@ async def base(self, request: web.Request) -> Dict[str, Any]:
         """Return network information."""
         return {
             ATTR_UNSUPPORTED: self.sys_resolution.unsupported,
-            ATTR_SUGGESTIONS: self.sys_resolution.suggestions,
-            ATTR_ISSUES: self.sys_resolution.issues,
+            ATTR_SUGGESTIONS: [
+                attr.asdict(suggestion)
+                for suggestion in self.sys_resolution.suggestions
+            ],
+            ATTR_ISSUES: [attr.asdict(issue) for issue in self.sys_resolution.issues],
         }
 
     @api_process
     async def apply_suggestion(self, request: web.Request) -> None:
         """Apply suggestion."""
         try:
-            suggestion = Suggestion(request.match_info.get("suggestion"))
+            suggestion = self.sys_resolution.get_suggestion(
+                request.match_info.get("suggestion")
+            )
             await self.sys_resolution.apply_suggestion(suggestion)
-        except ValueError:
-            raise APIError("Suggestion is not valid") from None
+        except ResolutionNotFound:
+            raise APIError("Suggestion uuid is not valid") from None
 
     @api_process
     async def dismiss_suggestion(self, request: web.Request) -> None:
         """Dismiss suggestion."""
         try:
-            suggestion = Suggestion(request.match_info.get("suggestion"))
+            suggestion = self.sys_resolution.get_suggestion(
+                request.match_info.get("suggestion")
+            )
             await self.sys_resolution.dismiss_suggestion(suggestion)
-        except ValueError:
-            raise APIError("Suggestion is not valid") from None
+        except ResolutionNotFound:
+            raise APIError("The supplied UUID is not a valid suggestion") from None
+
+    @api_process
+    async def dismiss_issue(self, request: web.Request) -> None:
+        """Dismiss issue."""
+        try:
+            issue = self.sys_resolution.get_issue(request.match_info.get("issue"))
+            await self.sys_resolution.dismiss_issue(issue)
+        except ResolutionNotFound:
```

Comment:
We should make it unique and raise an IssueNotFound exception

Reply:
Since we have no reason to make a difference, I don't think we should blow the number of our exceptions more as needed. If the time comes and we need that on the code base, I'm totally fine to make this